File: Task_2/task_2.php
<?php

function isPangram() {
    echo "Введите строку: ";
    $sentence = trim(fgets(STDIN));
    $sentence = strtolower($sentence);

    $letters = [];
    for ($i = 0; $i < strlen($sentence); $i++) {
        $char = $sentence[$i];
        if ($char >= 'a' && $char <= 'z') {
            $letters[$char] = true;
        }
    }
    return count($letters) == 26;
}

if (isPangram()) {
    echo "true\n";
} else {
    echo "false\n";
}
?>

File: Task_2/task_2.go
package main

import (
	"bufio"
	"fmt"
	"os"
	"strings"
	"unicode"
)

func isPangram() bool {
	reader := bufio.NewReader(os.Stdin)
	fmt.Print("Введите строку: ")
	sentence, _ := reader.ReadString('\n')
	sentence = strings.TrimSpace(sentence)
    sentence = strings.ToLower(sentence)

	letters := make(map[rune]bool)
	for _, char := range sentence {
        lowerChar := unicode.ToLower(char)
		if lowerChar >= 'a' && lowerChar <= 'z' {
			letters[lowerChar] = true;
		}
	}
	return len(letters) == 26
}

func main() {
	if isPangram() {
		fmt.Println("true")
	} else {
		fmt.Println("false")
	}
}
